### ISSUE TITLE
Unable to edit images

### ISSUE THREAD

**Issue by k2soule** [isawnyu/isaw.web#348]
I'm getting a blank dialog box this afternoon when I attempt to edit an image. It looks like this:
<img width="985" alt="screen shot 2018-10-30 at 4 57 29 pm" src="https://user-images.githubusercontent.com/7882896/47750546-00067c00-dc66-11e8-8be3-ac5886561082.png">

I am using Firefox (Mac).

**Comment by jessesnyder** [comment]
I'm getting a different error on Chrome/mac, where it's not possible to select and existing image, and there's a JS error in the console:
```
Uncaught DOMException: Failed to execute 'setBaseAndExtent' on 'Selection': There is no child at offset 1.
    at Editor.eval (eval at <anonymous> (http://isaw.nyu.edu/portal_javascripts/NYU%20ISAW%20Theme/resourceplone.app.jquery-cachekey-41bae1c8847cb60ec052f001660068dd.js:12:11369), <anonymous>:1:15386)
    at Dispatcher.dispatch (eval at <anonymous> (http://isaw.nyu.edu/portal_javascripts/NYU%20ISAW%20Theme/resourceplone.app.jquery-cachekey-41bae1c8847cb60ec052f001660068dd.js:12:11369), <anonymous>:1:6059)
    at DOMUtils.c (eval at <anonymous> (http://isaw.nyu.edu/portal_javascripts/NYU%20ISAW%20Theme/resourceplone.app.jquery-cachekey-41bae1c8847cb60ec052f001660068dd.js:12:11369), <anonymous>:1:185038)
    at j (eval at <anonymous> (http://isaw.nyu.edu/portal_javascripts/NYU%20ISAW%20Theme/resourceplone.app.jquery-cachekey-41bae1c8847cb60ec052f001660068dd.js:12:11369), <anonymous>:1:58686)
    at HTMLDocument.y (eval at <anonymous> (http://isaw.nyu.edu/portal_javascripts/NYU%20ISAW%20Theme/resourceplone.app.jquery-cachekey-41bae1c8847cb60ec052f001660068dd.js:12:11369), <anonymous>:1:58844)
```

**Comment by jessesnyder** [comment]
Bizarrely, Firefox on Mac is working OK for me.
![screen shot 2018-10-30 at 4 15 52 pm](https://user-images.githubusercontent.com/63560/47756413-2a087000-dc5f-11e8-9e6e-5d08f4086cef.png)

**Comment by jessesnyder** [comment]
The Chrome bug is this one, I think: https://github.com/tinymce/tinymce/issues/3611
Bug with TinyMCE, fixed long ago, but of course the Plone version lags, especially the version for Plone 4. 

@k2soule Can you open the browser console in Firefox and find what error is being thrown there, if any? I'm somewhat stuck on that since I can't reproduce it.

**Comment by jessesnyder** [comment]
For what it's worth, it also seems to work in Safari.

**Comment by k2soule** [comment]
I'm getting two warnings in the console:

Use of Mutation Events is deprecated. Use MutationObserver instead.
resourceplone.app.jquery-cachekey-365ab9692fb96369c8a405af247aa73c.js line 12 > eval:1:81613

Blocked loading mixed active content “http://fonts.googleapis.com/css?family=Questrial”[Learn More]    @@edit-slide

**Comment by jessesnyder** [comment]
Huh.. I don't get those particulary warnings (I get many others from Firefox), but if they're just warnings they wouldn't explain the error anyway.

**Comment by paregorios** [comment]
I am seeing this problem with any content items normally displayed as images in the insert/edit dialogue (images, exhibition objects. This happens when the dialogue is invoked to add/edit images or to add/edit links. I am using FF 63.0.1 (64-bit) under OSX El Capitan. I do **not** see any relevant console errors when loading a page in "edit" view or when launching the insert/edit dialog. 

The insert/edit dialog acts like it just isn't seeing the contained items:

![screen shot 2018-11-02 at 11 42 59 am](https://user-images.githubusercontent.com/263285/47928477-7983b180-de94-11e8-8f26-74893c50e530.png)

Note that folders and pages seem to show up fine:

![screen shot 2018-11-02 at 11 43 45 am](https://user-images.githubusercontent.com/263285/47928513-95875300-de94-11e8-921f-2eea592caade.png)

I will try to test in other browsers after my noon central meeting. 

I am also bumping this ticket to "urgent" as the inability to make links and image modifications hamstrings most of our content curation activities for our users.

**Comment by paregorios** [comment]
Clarification: my previous comment was informed by testing on PRODUCTION. I have not compared behavior on staging so I do not know if it is also happening there.

**Comment by k2soule** [comment]
Two updates:
I am not seeing the same behavior on the staging server. I am able to edit images and links in that environment.

I was able to edit images on production using Safari, so at least I have a workaround for immediate concerns.

**Comment by paregorios** [comment]
I am seeing intermittent behavior with images on production. I am trying to produce a repeatable workflow with exhibition objects.

**Comment by paregorios** [comment]
Steps to reproduce with an exhibition object:

1. Log into production
2. Navigate to an unobtrusive context.
3. Create [a folder entitled "Test 20181102"](https://isaw.nyu.edu/members/te20-40nyu.edu/test_20181102/)
4. Inside "Test 20181102", create a page entitled "Test Page" and put some arbitrary text in the Body Text field. Save. 
5. Inside "Test 20181102", create an exhibition object entitiled "Test Exhibition Object". Save.
6. Inside "Test 20181102", create and upload an image entitled "Test Image". Save.
7. Edit "Test Page." 
   - Select some of the text in the body field.
   - Select the "Insert/Edit link icon in the TinyMCE toolbar"
   - Notice that "Test Page" and "Test Image" are available in the default context, but not "Test Exhibition Object."
   - Cancel your edit.
8. Submit "Test Exhibition Object" for publication review.
9. Repeat step 7. Notice there is no change. Cancel your edit.
10. Publish "Test Exhibition Object".
11. Repeat step 7. Notice there is no change. Cancel your edit.

**Comment by paregorios** [comment]
For exhibition objects, I wonder if this is the problem: https://stackoverflow.com/questions/22799301/new-dexterity-content-type-is-not-visible-from-the-add-link-dialogue

**Comment by jessesnyder** [comment]
I agree that they sound different and warrant separate tickets.

**Comment by k2soule** [comment]
Ok, I think I've exterminated the gremlin in the case of the disappearing content in Insert/Edit boxes.
I tried disabling my ad blocker, but that didn't help.
Restarting the browser didn't help, refreshing the browser didn't help.
Then, I tried a new Private Browsing window, and it worked fine.
Manually cleared all cache & cookies, and now I can use a regular (non-private) firefox browsing window, too.
I'm downgrading the priority on this ticket, and if Tom splits out his other issue, I think we can close this one.

So... what the heck was it caching so stubbornly, to cause that weirdness for the past few days?
